### DISCUSSION TITLE
[expo-random] [expo-auth-session] Calling synchronous methods on native modules is not supported in Chrome

### DISCUSSION THREAD

**Discussion by lHersey** [expo/expo#13900] [Archived: Questions] [answered]
Hello everyone!

I'm trying to debug my app but there is a problem with a Native module.

```
export function getRandomBytes(byteCount) {
    assertByteCount(byteCount, 'getRandomBytes');
    const validByteCount = Math.floor(byteCount);
    if (ExpoRandom.getRandomBytes) {
        return ExpoRandom.getRandomBytes(validByteCount);
    }
    else if (ExpoRandom.getRandomBase64String) {
        const base64 = ExpoRandom.getRandomBase64String(validByteCount);
        return toByteArray(base64);
    }
    else {
        throw new UnavailabilityError('expo-random', 'getRandomBytes');
    }
} 
```

I'm getting `Calling synchronous methods on native modules is not supported in Chrome` when using `react-native-debugger`

`expo-random` is a peer dependency of `expo-auth-session`, that is why I have not tried to use `react-native-random-values`.

It works perfectly without the debugger enabled

Is there a way to make it work with `expo-random`?

Thanks!

**Comment by EvanBacon** [comment, marked as answer]
This should fix the issue when it lands https://github.com/expo/expo/pull/13616 the issue is related to remote debugging which runs the JS in chrome instead of on your device (i.e. sync calls are not supported).
